couchdb/server/compiler.py

```diff
@@ -78,3 +96,48 @@ def helper():
         'parent': mod,
         'id': (idx is not None) and (idx + '/' + name) or name
     })
+
+
+def import_b64egg(b64str, egg_cache=None):
+    """Imports top level namespace from base64 encoded egg file.
+
+    For Python 2.4 `setuptools <http://pypi.python.org/pypi/setuptools>`_
+    package required.
+
+    :param b64str: Base64 encoded egg file.
+    :type b64str: str
+
+    :return: Egg top level namespace or None if egg import disabled.
+    :rtype: dict
+    """
+    if iter_modules is None:
+        raise ImportError('No tools available to work with eggs.'
+                          ' Probably, setuptools package could solve'
+                          ' this problem.')
+    egg = None
+    egg_path = None
+    egg_cache = (egg_cache or
+                 os.environ.get('PYTHON_EGG_CACHE') or
+                 os.path.join(tempfile.gettempdir(), '.python-eggs'))
+    try:
+        try:
+            if not os.path.exists(egg_cache):
+                os.mkdir(egg_cache)
+            hegg, egg_path = tempfile.mkstemp(dir=egg_cache)
+            egg = os.fdopen(hegg, 'wb')
+            egg.write(base64.b64decode(b64str))
+            egg.close()
+            exports = EggExports(
+                [(name, loader.load_module(name))
+                 for loader, name, ispkg in iter_modules([egg_path])]
+            )
+        except:
+            log.exception('Egg import failed')
+            raise
+        else:
+            if not exports:
+                raise Error('egg_error', 'Nothing to export')
+            return exports
+    finally:
+        if egg_path is not None and os.path.exists(egg_path):
+            os.unlink(egg_path)
```

Comment:
May be, but there are no tools in python to load an egg from memory. At least there were not at those times, not sure what modern setuptools could do here today.